src/main/java/com/piekoszek/nowaksharedrent/auth/AuthController.java

```diff
@@ -15,18 +19,19 @@
 
     @PostMapping("/auth/signup")
     ResponseEntity<Object> createAccount(@RequestBody Account account) {
-        if(authService.createAccount(account)) {
-            return new ResponseEntity<>(new MessageResponse("Account created successfully"), HttpStatus.CREATED);
+        Optional<String> token = authService.createAccount(account);
+        if (token.isPresent()) {
+            return new ResponseEntity<>(new MessageResponse(token.get()), HttpStatus.CREATED);
         }
         return new ResponseEntity<>(new MessageResponse("Account with such email already exists"), HttpStatus.BAD_REQUEST);
     }
 
     @PostMapping("/auth/login")
-    ResponseEntity<Object> login(@RequestBody Account input) {
-        Account account = authService.findAccount (input.getEmail());
-        if (account != null && input.getPassword().equals(account.getPassword())) {
-            return new ResponseEntity<>(new MessageResponse("Login successful!"), HttpStatus.OK);
+    ResponseEntity<Object> login(@RequestBody Account account) {
+        Optional<String> token = authService.loginUser(account);
+        if (token.isPresent()) {
+            return new ResponseEntity<>(new MessageResponse(token.get()), HttpStatus.CREATED);
```

Comment:
Hmm maybe it should be sth else than MessageResponse, 
looks a bit wired, but up to You.
![image](https://user-images.githubusercontent.com/10707837/54352602-517e3900-4652-11e9-8778-dc9670c56b01.png)